docs(Grid): Add documentation about AI features (#3165)
telerik/blazor-docs: common-features/integration-with-telerik-ai-smartcomponents-extensions.md

---
title: Integration with Telerik.AI.SmartComponents.Extensions
page_title: Integration with Telerik.AI.SmartComponents.Extensions
description: How to integrate the UI for Blazor components with Telerik.AI.SmartComponents.Extensions
slug: common-features-telerik-ai-smartcomponents-extensions-integration
tags: telerik,blazor,aiprompt,ai,extensions,integration
published: True
position: 45
---

# Getting Started with Telerik.AI.SmartComponents.Extensions

## Overview

The `Telerik.AI.SmartComponents.Extensions` library provides AI-powered functionality for Grid operations, enabling natural language processing for filtering, sorting, grouping, and highlighting data. This library integrates seamlessly with Microsoft.Extensions.AI and Azure OpenAI to provide intelligent Grid interactions.

## Prerequisites

- .NET 8.0 or later
- `Microsoft.Extensions.AI` package
- Azure OpenAI or OpenAI API access
- ASP.NET Core (for web API scenarios)

## Installation

> Add the `Telerik.AI.SmartComponents.Extensions` NuGet package to your Blazor app. The package is hosted on `nuget.org`.
> 
> Add the required dependency packages:
> 
> * `Microsoft.Extensions.AI`
> * `Azure.AI.OpenAI`

Author: ```suggestion
1. Add the [`Telerik.AI.SmartComponents.Extensions` NuGet package](https://www.nuget.org/packages/Telerik.AI.SmartComponents.Extensions) to your project.
1. Add the required dependency packages:
    * `Microsoft.Extensions.AI`
    * `Azure.AI.OpenAI`
```

### 1. Column Configuration

When the options for the column are of Enum type provide meaningful column values to help the AI understand your data:

```csharp
new GridAIColumn 
{ 
    Field = "Status", 
    // only when only a set of values are used
    Values = new[] { "Active", "Inactive", "Pending" } 
}
```

### 2. Error Handling

```csharp
try
{
    var completion = await chatService.CompleteChatAsync(messages, options);
    var response = completion.ToolCalls.ExtractGridOptions();
    return Json(response);
}
catch (Exception ex)
{
    return StatusCode(500, $"AI processing failed: {ex.Message}");
}
```

### 3. Input Validation

```csharp
if (request?.Columns == null || !request.Columns.Any())
{
    return BadRequest("Columns are required");
}

if (request.Contents == null || !request.Contents.Any())
{
    return BadRequest("Content is required");
}
```

## Testing

The library includes comprehensive test coverage. You can run tests to verify functionality:

```bash
cd tests
dotnet test
```

For integration testing with your specific data model, create test cases that verify AI responses match expected Grid operations.

## Troubleshooting

### Common Issues

> * Invalid URI Format: Ensure your Azure OpenAI endpoint is correctly formatted in configuration
> * API Key Issues: Verify your API key has proper permissions and is not expired
> * Model Availability: Ensure the specified model ID is deployed in your Azure OpenAI resource
> * Token Limits: Be mindful of token limits when processing large datasets

Author: ```suggestion
* Invalid URI Format: Ensure your Azure OpenAI endpoint is correctly formatted in configuration.
* API Key Issues: Verify your API key has proper permissions and is not expired.
* Model Availability: Ensure the specified model ID is deployed in your Azure OpenAI resource.
* Token Limits: Be mindful of token limits when processing large datasets.
```

> 1. **Invalid URI Format**: Ensure your Azure OpenAI endpoint is correctly formatted in configuration
> 2. **API Key Issues**: Verify your API key has proper permissions and is not expired
> 3. **Model Availability**: Ensure the specified model ID is deployed in your Azure OpenAI resource
> 4. **Token Limits**: Be mindful of token limits when processing large datasets

Author: ```suggestion
* Invalid URI Format: Ensure your Azure OpenAI endpoint is correctly formatted in configuration
* API Key Issues: Verify your API key has proper permissions and is not expired
* Model Availability: Ensure the specified model ID is deployed in your Azure OpenAI resource
* Token Limits: Be mindful of token limits when processing large datasets
```
